core/src/transfer/moonlight.rs

```diff
@@ -274,6 +302,25 @@ impl Transaction {
         Some(stripped_transaction)
     }
 
+    /// Creates a modified clone of this transaction if it contains a Blob,
+    /// clones all fields except for the Blob, where its hash is set as Memo.
+    ///
+    /// Returns none if the transaction is not a Blob transaction.
+    #[must_use]
+    pub fn convert_blob(&self) -> Option<Self> {
+        let data = self.data()?;
+
+        if let TransactionData::Blob(_) = data {
+            let hash = data.signature_message();
```

Comment:
shouldn't you hash the `data.signature_message()` (proposed name `data.to_hash_input_bytes`) before storing it in the memo?

Reply:
the hashing is done by `signature_message`. which is why this is not `to_hash_input_bytes` and I still think this function actually strips-off the blob (and additionally sets the hash as Memo) rather than converting it to Memo.